php-checks/src/main/java/org/sonar/php/checks/PHP5DeprecatedFunctionUsageCheck.java

```diff
@@ -63,56 +68,53 @@
     .put("mysql_escape_string", "mysql_real_escape_string")
     .build();
 
+  private static final String SET_LOCALE_FUNCTION = "setlocale";
   private static final ImmutableSet<String> LOCALE_CATEGORY_CONSTANTS = ImmutableSet.of(
     "LC_ALL", "LC_COLLATE", "LC_CTYPE", "LC_MONETARY", "LC_NUMERIC", "LC_TIME", "LC_MESSAGES");
 
   @Override
-  public void init() {
-    subscribeTo(PHPGrammar.MEMBER_EXPRESSION);
+  protected ImmutableSet<String> functionNames() {
+    return ImmutableSet.<String>builder()
+      .addAll(NEW_BY_DEPRECATED_FUNCTIONS.keySet())
+      .add(SET_LOCALE_FUNCTION)
+      .build();
   }
 
   @Override
-  public void visitNode(AstNode astNode) {
-    AstNode memberExpr = astNode.getFirstChild();
-    AstNode nextNode = memberExpr.getNextAstNode();
-    String calledFunctionName = memberExpr.getTokenOriginalValue();
-    String replacement = NEW_BY_DEPRECATED_FUNCTIONS.get(calledFunctionName);
-
-
-    if (nextNode != null && nextNode.is(PHPGrammar.FUNCTION_CALL_PARAMETER_LIST)) {
+  protected void createIssue(FunctionCallTree tree) {
+    String functionName = functionName(tree.callee());
 
-      if ("setlocale".equals(calledFunctionName)) {
-        String category = getCategoryFromStringLiteralParameter(nextNode);
+    if (SET_LOCALE_FUNCTION.equals(functionName)) {
+      checkLocalCategoryArgument(tree, tree.arguments());
 
-        if (category != null) {
-          getContext().createLineViolation(this, "Use the \"{0}\" constant instead of a string literal.", astNode, category);
-        }
-
-      } else if (replacement != null) {
-        getContext().createLineViolation(this, buildMessage(calledFunctionName, replacement), astNode);
-      }
+    } else {
+      context().newIssue(this.KEY, buildMessage(functionName)).tree(tree.callee());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this instance-reference to a static reference. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2209)